src/main/java/com/influxdb/v3/client/internal/InfluxDBClientImpl.java

```diff
@@ -421,7 +414,24 @@ private Stream<VectorSchemaRoot> queryData(@Nonnull final String query,
             }
         });
 
-        CallOption[] callOptions = options.grpcCallOptions().getCallOptions();
+        GrpcCallOptions.Builder builder = new GrpcCallOptions.Builder()
+            .fromGrpcCallOptions(options.grpcCallOptions());
+
+        if (options.grpcCallOptions().getDeadline() == null) {
+            if (config.getQueryTimeout() != null) {
+                builder.withDeadline(Deadline.after(config.getQueryTimeout().toMillis(), TimeUnit.MILLISECONDS));
+            }
+        } else if (options.grpcCallOptions().getDeadline().timeRemaining(TimeUnit.NANOSECONDS) <= 0) {
+            LOG.warning("Received impractical gRPC call options deadline "
+                + options.grpcCallOptions().getDeadline());
+            if (config.getQueryTimeout() != null) {
+                LOG.warning("Using configuration query timeout "
+                    + config.getQueryTimeout() + " as a replacement.");
+                builder.withDeadline(Deadline.after(config.getQueryTimeout().toMillis(), TimeUnit.MILLISECONDS));
+            }
+        }
```

Comment:
What about something simple like:


```suggestion
       if (config.getQueryTimeout() != null) {
            builder.withDeadline(Deadline.after(config.getQueryTimeout().toMillis(), TimeUnit.MILLISECONDS));
        } else if (options.grpcCallOptions().getDeadline() != null) {
            LOG.warning("No query timeout configured in ClientConfig, but gRPC call options contains a deadline "
                    + options.grpcCallOptions().getDeadline()
                    + ". This may lead to unexpected behavior - use setting query timeout in ClientConfig.");
        }
```

?